src/Pmi/Service/PatientStatusService.php

```diff
@@ -22,35 +25,75 @@ public function sendPatientStatusToRdr()
     {
         $limit = $this->app->getConfig('patient_status_queue_limit');
         $query = "
-            SELECT psh.id,
-                   psh.patient_status_id,
-                   psh.site,
-                   psh.status,
-                   psh.comments,
-                   psh.created_ts as authored,
-                   ps.participant_id,
-                   ps.organization,
-                   ps.awardee,
+            SELECT psir.*,
+                   psi.site,
+                   psi.created_ts as authored,
+                   psi.organization,
+                   psi.awardee,
+                   ps.id as patient_status_id,
+                   u.id as user_id,
                    u.email as user_email
-            FROM patient_status_history psh
-            INNER JOIN patient_status ps ON ps.id = psh.patient_status_id
-            LEFT JOIN users u ON psh.user_id = u.id
-            WHERE psh.rdr_ts is null
-            ORDER BY psh.id ASC
+            FROM patient_status_import_rows psir
+            INNER JOIN patient_status_import psi ON psi.id = psir.import_id AND psi.confirm = :confirm
+            LEFT JOIN patient_status ps ON ps.participant_id = psir.participant_id AND psi.organization = ps.organization
+            LEFT JOIN users u ON psi.user_id = u.id
+            WHERE psir.rdr_status = :rdrStatus
+            ORDER BY psir.id ASC
             limit {$limit}
```

Comment:
I know this was already here before, but can we use a parameter for the limit? While we have control over the config items, we should treat them like untrusted input.

Reply:
Somehow setting parameter to a limit clause throws a syntax error. It may be related to this https://github.com/doctrine/dbal/issues/3459

I tried setting it to `$patientStatuses = $this->em->fetchAll($query, ['confirm' => 1, 'rdrStatus' => 0, 'limit' => $limit], [\PDO::PARAM_INT, \PDO::PARAM_INT, \PDO::PARAM_INT]);` but that didn't work either.

Reply:
Oh, right. We should just cast the setting to an integer. And maybe abort if it is < 1. No parameter needed in that case.